server/engine/src/main/java/org/eclipse/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -911,7 +911,15 @@ public List<Node> visitStartStatement(StartStatementContext ctx) {
 
   @Override
   public List<Node> visitPerformInlineStatement(PerformInlineStatementContext ctx) {
-    return addTreeNode(ctx, PerformNode::new);
+      boolean untilExit = Optional.of(ctx)
+          .map(PerformInlineStatementContext::performType)
+          .map(PerformTypeContext::performUntil)
+          .map(PerformUntilContext::condition)
+          .map(ConditionContext::getStart)
+          .map(Token::getText)
+          .map("EXIT"::equals)
```

Comment:
I think this is case sensitive.

Also I think it would better to adjust the grammar to support it directly e.g.:
```
performUntil
   : performTestClause? UNTIL (EXIT | condition)
   ;
```